csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -154,11 +179,31 @@ public override void Dispose()
             }
         }
 
-        internal static async Task<TGetResultSetMetadataResp> GetResultSetMetadataAsync(TOperationHandle operationHandle, TCLIService.IAsync client, CancellationToken cancellationToken = default)
+        internal static async Task<TGetResultSetMetadataResp> GetResultSetMetadataAsync(TOperationHandle operationHandle, TCLIService.IAsync client, ActivitySource? activitySource = null, CancellationToken cancellationToken = default)
         {
+            using var activity = StartActivity(activitySource, nameof(GetResultSetMetadataAsync));
             TGetResultSetMetadataReq request = new(operationHandle);
             TGetResultSetMetadataResp response = await client.GetResultSetMetadata(request, cancellationToken);
             return response;
         }
+
+        internal static void TraceException(Exception exception, Activity? activity, bool escaped = true)
```

Comment:
TODO: This instrumentation should be available in common to all driver implementation.